src/edu/washington/escience/myria/expression/evaluate/GenericEvaluator.java

```diff
@@ -121,27 +131,93 @@ public String getJavaExpressionWithAppend() {
   }
 
   /**
+   * Holder class for results and result counts from {@link #evaluateColumn}.
+   */
+  public static class EvaluatorResult {
+    private final ReadableColumn results;
+    private final Column<?> resultCounts;
+
+    protected EvaluatorResult(
+        @Nonnull final ReadableColumn results, @Nonnull final Column<?> resultCounts) {
+      this.results = results;
+      this.resultCounts = resultCounts;
+    }
+    /**
+     * @return a {@link ReadableColumn} containing results from {@link #evaluateColumn}
+     */
+    public ReadableColumn getResults() {
```

Comment:
Not sure what you mean by that; we can only pass `ReadableColumn` to `appendFromColumn()`. Actually, I'm finding that returning `List<Column>` from `getResults()` is making things considerably more awkward in `Apply.fetchNextReady()`, because I now have to do my own row index management for `List<Column>` instead of letting `TupleBuffer` do it for me.